x-pack/plugins/ml/public/data_frame/common/index.ts

```diff
@@ -134,11 +159,38 @@ export function getDataFramePreviewRequest(
   };
 
   groupBy.forEach(g => {
-    request.pivot.group_by[g] = {
-      terms: {
-        field: g,
-      },
-    };
+    if (g.agg === PIVOT_SUPPORTED_GROUP_BY_AGGS.TERMS) {
+      const termsAgg: TermsAgg = {
+        terms: {
+          field: g.field,
+        },
+      };
+      request.pivot.group_by[g.formRowLabel] = termsAgg;
+    } else if (g.agg === PIVOT_SUPPORTED_GROUP_BY_AGGS.HISTOGRAM) {
+      const histogramAgg: HistogramAgg = {
+        histogram: {
+          field: g.field,
+          interval: g.interval,
+        },
+      };
+      request.pivot.group_by[g.formRowLabel] = histogramAgg;
+    } else if (g.agg === PIVOT_SUPPORTED_GROUP_BY_AGGS.DATE_HISTOGRAM) {
+      const dateHistogramAgg: DateHistogramAgg = {
+        date_histogram: {
+          field: g.field,
+          interval: g.interval,
+        },
+      };
+
+      const timeUnitMatch = g.interval.match(dateHistogramIntervalFormatRegex);
+      if (timeUnitMatch !== null && Array.isArray(timeUnitMatch) && timeUnitMatch.length === 2) {
```

Comment:
Might be worth an extra line in the comment here, to clarify we are checking for a match like `1d` for 1 day, with two parts - a numeric part and an interval part.

Reply:
Added another comment, also added another safety check to not add the format if it's `undefined` (which could happen for `w`). Done in e851cdd.